event-buffer/event-buffer-core/src/main/java/uk/gov/justice/services/event/buffer/core/repository/spliterator/EventBufferSpliterator.java

```diff
@@ -0,0 +1,44 @@
+package uk.gov.justice.services.event.buffer.core.repository.spliterator;
+
+import uk.gov.justice.services.event.buffer.core.repository.streambuffer.StreamBufferEvent;
+
+import java.util.Iterator;
+import java.util.Spliterator;
+import java.util.Spliterators;
+import java.util.function.Consumer;
+
+public class EventBufferSpliterator {
+    private final long currentVersion;
+    private final Iterator<StreamBufferEvent> bufferEventIterator;
+
+    public EventBufferSpliterator(long currentVersion, Iterator<StreamBufferEvent> bufferEventIterator) {
+        this.currentVersion = currentVersion;
+        this.bufferEventIterator = bufferEventIterator;
+    }
+
+    public Spliterators.AbstractSpliterator<StreamBufferEvent> invoke() {
+        return new Spliterators.AbstractSpliterator<StreamBufferEvent>(
+                Long.MAX_VALUE, Spliterator.ORDERED) {
+            long statusVersion = currentVersion;
```

Comment:
Is there a reason we need to reassign this to a local variable?
Presumably this isn't a singleton